### ISSUE TITLE
"Bus error" when using --onefile on Raspbery Pi 4 32bit Linux (Buster)

### ISSUE THREAD

**Issue by gaamaaresosa** [Nuitka/Nuitka#2238]
Bus error !

I tried both Nuitka 1.5.7 and Nuitka 1.5.8

I compiled a python 3 project with below command in Raspberry Pi 4B 32bit OS.
I got the output successfully
But when I execute it returns below line and quit.

"Bus error"

The Nuitka command:
sudo python3 -m nuitka --standalone --onefile --follow-imports --include-data-dir=templates=templates --output-filename=MyExecutable MyExecutable.py

Similar issue was there in PyInstaller 5.10.0 on Apr 14th 2023
Which I had raised and they have fixed in few hours.

Bus error in latest PyInstaller 5.10.0 on Raspberry Pi 32bit Buster OS #7566
https://github.com/pyinstaller/pyinstaller/issues/7566

archive: fix TOC entry alignment in PKG/CArchive #7571
https://github.com/pyinstaller/pyinstaller/pull/7571

Similar issue I face now with Nuitka too.
Could you please fix this.

My System:
Raspberry Pi 4B
OS: Buster Linux 32Bit

**Comment by kayhayen** [comment]
I am not sure, I want to read through issue reports of other projects, just to find out, what the issue is. Can you provide a minimal reproducer, so I can see this for myself?

**Comment by gaamaaresosa** [comment]
> I am not sure, I want to read through issue reports of other projects, just to find out, what the issue is. Can you provide a minimal reproducer, so I can see this for myself?

Sure !
kindly create a simple python test file named: BusErrorTest.py
```
#!/usr/bin/python3
print('This is a PyInsyaller 5.10 Bus error test')
```
Then compile it with any latest Nuitka
```
sudo python3 -m nuitka --standalone --onefile BusErrorTest.py -o BusErrorTest
```

Finally execute the BusErrorTest file on only the 
32bit Linus 10 (Buster) May be on Raspberry Pi - 4
Or on any 32bit Linus 10 (Buster) container 

You will be
Bus error

Similar issue was in PyInsytaller latest version and they have fixed it.
https://github.com/pyinstaller/pyinstaller/issues/7566

archive: fix TOC entry alignment in PKG/CArchive #7571
https://github.com/pyinstaller/pyinstaller/pull/7571

Please fix this in your Nuitka also.
I found Nuitka is the best as it protect our source code from reverse enginnering.

Thanks a ton Keyhayen !

**Comment by kayhayen** [comment]
@gaamaaresosa using 32 bit in these days is clearly a sign of being commercial, you may want to consider subscribing to: https://nuitka.net/doc/commercial.html to give this any kind of priority. 

The issue appears to be that accessing short and long will only work on boundaries. And the constants' loader code and the onefile bootstrap may both be affected. Clearly doable, but I believe I cannot do it without funding. Maybe somebody else will volunteer to do it. 

I did not immediately find the mistake, where Nuitka ever casts any structure, long or int, to memory. We just don't do it that way. So that's where I will have to leave it then. The fix from that other project doesn't apply to Nuitka.

**Comment by gaamaaresosa** [comment]
> @gaamaaresosa using 32 bit in these days is clearly a sign of being commercial

Just to clarify your doubt, I would have prefered 64bit for commercial application rathen than prefering obsoleted 32bit. This is an existing old  school automated attendence project. As they wanted to protect they source now. I still try convince them for your commecial scheme too.

> I believe I cannot do it without funding. Maybe somebody else will volunteer to do it.

Ok, time being I will manage.

> The fix from that other project doesn't apply to Nuitka.

Agreed.

**Comment by kayhayen** [comment]
I have a large set of customers that go with Nuitka commercial and work on systems as old Windows XP with Python 3.4, or RHEL 5, and I will be damned to form an opinion of that, but working in Air Traffic Control for 20 years, I have a bunch of ancient hardware and therefore OSes that is irreplaceable unless through projects that last half a decade if it's quick. Customers really like to just add one more thing.

However, I of course have to be conservative with my time. I cannot spend 1 day, which is a substantial part of my time, for something that benefits only a small subset of users, that then may very well decide that free Nuitka is good enough for them with that.

**Comment by gaamaaresosa** [comment]
Yes I agree some time one can't upgrade to 64bit as some old libraries will not support latest 64bit.
But I got a request from our old school master as he suspects some of their current students hacking their auto attendence register.
Even I have to try that on latest 64bit OS (Raspberry Pi)

And I aggree already you had done wornderful unique job in Nuitka.
Spending time on it again should give you some more benifits.

Thanks again !
